### PR TITLE
[peggy] list missed cosmos event

### PR DESCRIPTION
**Pull request by juniuszhou**
the pr for new sub command to list all missed cosmos events, the ebrelayer not process them. 
the command usage:     
ebrelayer listMissedCosmosEventCmd [tendermintNode] [web3Provider] [bridgeRegistryContractAddress] [days]

### CODE REVIEW

**Review by ElliotFriedman — APPROVED** [review]

**Review by ElliotFriedman — CHANGES_REQUESTED** [review]
I ran `ebrelayer listMissedCosmosEventCmd tcp://52.205.192.58:26657 wss://mainnet.infura.io/ws/v3/infura-id 0x5c350c858035a5dc8a64FAa52a13BB926b1dD0Bd 0x0D7dEF5C00a8B6ddc58A0255f0a94cc739C6d0B5 1` and it gave me this error:

`ERROR: Must specify a 'chain-id'`

I would appreciate if you could provide a bit more docs around how to run this command.

File: cmd/ebrelayer/relayer/cosmos.go [hunk 1]
```diff
@@ -175,7 +155,7 @@ func (sub CosmosSub) getAllProphecyClaim(ethFromBlock int64, ethToBlock int64) [
 			}
 
 			// compare tx sender with my ethereum account
-			if sender != mySender {
+			if sender != ethereumAddress {
```
Comment (ElliotFriedman) [inline cmd/ebrelayer/relayer/cosmos.go:158]: Running this locally with the following command,

`ebrelayer listMissedCosmosEventCmd tcp://52.205.192.58:26657 wss://mainnet.infura.io/ws/v3/<infuraid> 0x5c350c858035a5dc8a64FAa52a13BB926b1dD0Bd 0x0D7dEF5C00a8B6ddc58A0255f0a94cc739C6d0B5 1`

I got this error which seems to be an issue on line 164 of cosmos.go:

cosmos.go:140: getAllProphecyClaim current blockNumber is 11967672
panic: runtime error: slice bounds out of range [:4] with capacity 0

goroutine 1 [running]:
github.com/Sifchain/sifnode/cmd/ebrelayer/relayer.GetAllProphecyClaim(0xc000010ae8, 0xddb6a8005cef7d0d, 0xc74ca9f055028ac5, 0xb5d0c639, 0xb69af7, 0xb6b177, 0x0, 0x100ffffffffffff, 0x0)
        /private/tmp/sifnode/cmd/ebrelayer/relayer/cosmos.go:164 +0x9f5
github.com/Sifchain/sifnode/cmd/ebrelayer/relayer.ListMissedCosmosEvent.ListMissedCosmosEvent(0x7ffeefbff72b, 0x19, 0x7ffeefbff745, 0x3e, 0xdca53580850c355c, 0x92bb132aa5fa648a, 0x5cef7d0dbdd01d6b, 0x55028ac5ddb6a800, 0xb5d0c639c74ca9f0, 0x1)
        /private/tmp/sifnode/cmd/ebrelayer/relayer/listMissedEvent.go:61 +0x1b7
main.RunListMissedCosmosEventCmd(0xc0010cd080, 0xc0002ba8c0, 0x5, 0x5, 0x0, 0x0)
        /private/tmp/sifnode/cmd/ebrelayer/replay.go:205 +0x405
github.com/spf13/cobra.(*Command).execute(0xc0010cd080, 0xc0002ba870, 0x5, 0x5, 0xc0010cd080, 0xc0002ba870)
        /Users/elliotfriedman/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:850 +0x47c
github.com/spf13/cobra.(*Command).ExecuteC(0x5969100, 0x447d605, 0x5969100, 0x4e935fa)
        /Users/elliotfriedman/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/elliotfriedman/go/pkg/mod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/tendermint/tendermint/libs/cli.Executor.Execute(0x5969100, 0x504a5f8, 0x9, 0xc000ed0450)
        /Users/elliotfriedman/go/pkg/mod/github.com/tendermint/tendermint@v0.33.9/libs/cli/setup.go:89 +0x3c
main.main()
        /private/tmp/sifnode/cmd/ebrelayer/main.go:242 +0xa7

**Review by ElliotFriedman — CHANGES_REQUESTED** [review]

File: cmd/ebrelayer/relayer/cosmos.go (hunk 1 above)
Comment (ElliotFriedman) [reply cmd/ebrelayer/relayer/cosmos.go:158]: Ran it again, and got the same issue.

Comment (juniuszhou) [reply cmd/ebrelayer/relayer/cosmos.go:159]: the tx is not smart contract call, will add protection for slice operation.

**Review by ElliotFriedman — APPROVED** [review]

**Pull request merged**